src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -60,200 +114,837 @@ public static void init()
+        {
+            Log.getLogger().error("Directory doesn't exist and failed to be created: " + directory.toString());
+        }
     }
 
     /**
-     * Returns a set of huts.
-     * This is the key set of {@link #hutStyleMap}.
+     * Get the list of Sections.
+     * Builder, Citizen, Farmer ... + decorations and scans.
      *
-     * @return Set of huts with a schematic.
+     * @return list of sections.
      */
-    public static Set<String> getHuts()
+    @NotNull
+    @SideOnly(Side.CLIENT)
+    public static List<String> getSections()
     {
-        return hutStyleMap.keySet();
+        final ArrayList<String> list = new ArrayList<>(schematicsMap.keySet());
+        Collections.sort(list);
+        return list;
     }
 
     /**
-     * Returns the max level for the provided hut.
+     * Get the list of styles for a given section.
      *
-     * @param hut Hut to get max level for.
-     * @return Max level of hut.
+     * @param section such as decorations, Builder ...
+     * @return the list of style for that section.
      */
-    public static int getMaxLevelForHut(final String hut)
+    @NotNull
+    @SideOnly(Side.CLIENT)
+    public static List<String> getStylesFor(final String section)
     {
-        return hutLevelsMap.getOrDefault(hut, 0);
+        if (schematicsMap.containsKey(section))
+        {
+            final Map<String, Map<String, String>> sectionMap = schematicsMap.get(section);
+            final ArrayList<String> list = new ArrayList<>(sectionMap.keySet());
+            Collections.sort(list);
+            return list;
+        }
+        return new ArrayList<>();
     }
 
     /**
-     * Returns a set of decorations.
-     * This is the key set of {@link #decorationStyleMap}.
+     * Get a list of schematics for this section and style.
      *
-     * @return Set of decorations with a schematic.
+     * @param section such as Builder, schematics. scans ...
+     * @param style limit the list for schematics to this style.
+     * @return the list of schematics
      */
-    public static Set<String> getDecorations()
+    @NotNull
+    @SideOnly(Side.CLIENT)
+    public static List<String> getSchematicsFor(final String section, final String style)
     {
-        return decorationStyleMap.keySet();
+        if (schematicsMap.containsKey(section))
+        {
+            final Map<String, Map<String, String>> sectionMap = schematicsMap.get(section);
+            if (sectionMap.containsKey(style))
+            {
+                final ArrayList<String> list = new ArrayList<>(sectionMap.get(style).values());
+                Collections.sort(list);
+                return list;
+            }
+        }
+        return new ArrayList<>();
     }
 
     /**
-     * Returns a list of styles for one specific decoration.
+     * Class to handle schematic naming.
+     * It does extract information from a schematic using its name.
+     */
+    public static class StructureName
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce this class from 118 to the maximum allowed 25 or externalize it in a public class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)